pxteditor/editor.ts

```diff
@@ -545,6 +545,9 @@ namespace pxt.editor {
         blockId?: string;
     }
 
+
+    export let upgradeXml: (code: string) => void;
```

Comment:
is there something preventing this upgrade from being inside the other xml upgrade code we already have? The ones returned by initExtensionsAsync?

Reply:
blocklyPatch

Reply:
oh yes initially it was because the dom we were getting there was for the workspace and not the tutorial.  Should I change it to also take in tutorial XML?

Reply:
the tutorial xml is also a workspace, it just isn't loaded in the editor (it gets rendered to an svg). i think it's fair to call the blocklypatch function on the decompiled tutorial xml as well.

Reply:
you shouldn't have to change the function though, just also call it in the tutorial loading code

Reply:
🎉 i will do that!